### DISCUSSION TITLE
https://teddyhuang-00.github.io/blog-using-pelican-and-github-pages.html

### DISCUSSION THREAD

**Discussion by giscus** [TeddyHuang-00/teddyhuang-00.github.io#3] [Announcements]
# https://teddyhuang-00.github.io/blog-using-pelican-and-github-pages.html

https://teddyhuang-00.github.io/blog-using-pelican-and-github-pages.html

**Comment by HuiwenZheng-Ariel** [comment]
:>>>>>

**Comment by TeddyHuang-00** [comment]
> 然后在 Makefile 中将 github 目标修改为：
> ```make
> github: publish
>     cp giscus.json "$(OUTPUTDIR)/"
>     ghp-import -m "Generate Pelican site $(shell date -u +"%Y-%m-%d %H:%M:%S")" -b $(GITHUB_PAGES_BRANCH) "$(OUTPUTDIR)"
>     git push origin $(GITHUB_PAGES_BRANCH)
> ```
> 这样的话，在执行一次 make github 之后 giscus.json 会存在于发布分支的根目录中，因此只有在 teddyhuang-00.github.io 这个域名下的页面才会加载 Giscus，本地预览的时候就不会加载了。有关 Giscus 更进阶的配置，可以参考官方文档。

研究了一下，其实Pelican完全可以拷贝一些内容到根目录下面，不需要这样手动操作，我们主要依赖 `pelicanconf.py` 中修改默认静态内容路径，然后指定静态内容路径映射到根目录即可：

具体来说，向 `pelicanconf.py` 中添加如下内容

```py
STATIC_PATHS = [
    "images", # default
    "extra", # 在 content 文件夹中新建的文件夹，也可以是其他名字，需要与下面保持一致
]
EXTRA_PATH_METADATA = {
    # 把 giscus.json 文件放在 extra 文件夹内，
    # 这边将其路径映射至个目录，这样在生成项目的时候就会自动从目录中把文件拷贝到路径下了！
    "extra/giscus.json": {"path": "giscus.json"},
}
```

就可以避免修改 `Makefile` 啦！这样可拓展性也更强一些，你可以将任意文件放到任何路径（如果有需要的话）
